### PR TITLE
Move resolveAllAdvanceSelection tests to gameLogic.test.js; document settings-gate exception

### PR DESCRIPTION
**Pull request by mikebud8**
## What changed

**`src/gameLogic.test.js`**
- Added `resolveAllAdvanceSelection` to the import list
- Appended its two tests at the end of the file, under a new `resolveAllAdvanceSelection` section

**`src/screens/VoteScreen.test.jsx`**
- Removed the `resolveAllAdvanceSelection` import (it's a pure function — nothing to do in a screen test file)
- Removed the `resolveAllAdvanceSelection` describe block (now in `gameLogic.test.js`)
- Added a labeled comment above the 4 remaining tests acknowledging them as intentional settings-gate smoke tests, explaining why they live here rather than in a `.js` test file

**`CLAUDE.md`**
- Extended the "Test what matters" bullet with an explicit exception for settings-gate smoke tests: what they are, why they're acceptable, and how to keep them scoped (one `describe` per screen, `renderToStaticMarkup` + `toContain` only, no interaction or state transitions)

## Test notes
- 429 tests, 0 failures
- `npm run lint`: 0 errors

### CODE REVIEW

File: src/gameLogic.test.js [hunk 1]
```diff
@@ -2536,3 +2537,22 @@ describe('computeSeasonToneDisplay', () => {
     expect(result.premise).toBe('has one')
   })
 })
+
```
[the remark this answers is not in the document]
Comment (mikebud8) [reply src/gameLogic.test.js:2540]: Dismissed — the function is a single ternary with two branches, both covered. There are no meaningful edge cases: allowMerges is a boolean, selectedIds is passed through unchanged, and the return shapes are fixed literals. Adding more tests would be padding, not coverage.
